### PR TITLE
Alerting: PagerDuty receiver to let user configure fields Source, Client and Client URL

### LINKED ISSUE #59502: PagerDuty Notifier - Source is set to os.Hostname
**What happened**:
When using the PagerDuty notifier, Grafana is hardcoded to use os.Hostname as the source.
https://github.com/grafana/grafana/blob/v9.2.6/pkg/services/ngalert/notifier/channels/pagerduty.go#L206

**What you expected to happen**:
PagerDuty API docs state that source should refer to ` The unique location of the affected system, preferably a hostname or FQDN.`

Although Grafana is the system generating the alert, in many cases it may not be the affected system the alert is referring to. 
In our case, we'd like to template the source field with Prometheus labels such as cluster, namespace, pod etc. Something actionable.

Its fair to say os.hostname could be a default when its source is empty but I don't think it should be hardcoded.

**Environment**:
- Grafana version: 9.2.6

### PR DESCRIPTION
**Pull request by yuri-tceretian**
**What is this feature?**
This pull request introduces support for new fields of PagerDuty configuration: source, client, and client_url. 

image

**Why do we need this feature?**
This is needed for the unification of Grafana and Alertmanager receivers.

**Which issue(s) does this PR fix?**:
Related: https://github.com/grafana/alerting/issues/20#issuecomment-1317895945

Fixes https://github.com/grafana/grafana/issues/59502

**Special notes for your reviewer**:

### CODE REVIEW

File: pkg/services/ngalert/notifier/channels/pagerduty.go [hunk 1]
```diff
@@ -23,13 +23,14 @@ const (
 	pagerDutyEventTrigger = "trigger"
 	pagerDutyEventResolve = "resolve"
 
-	defaultSeverity = "critical"
+	defaultSeverity = "error"
```
Comment (alexweav) [inline pkg/services/ngalert/notifier/channels/pagerduty.go:26]: I wonder if we should submit this line's change in a separate PR, with `add to changelog`? That way, this change gets called out specifically in the patch notes.

File: pkg/services/ngalert/notifier/channels/pagerduty.go [hunk 2]
```diff
@@ -88,7 +92,19 @@ func buildPagerdutySettings(fc FactoryConfig) (*pagerdutySettings, error) {
...
 	}
-
+	if settings.Client == "" {
+		settings.Client = defaultClient
+	}
+	if settings.ClientURL == "" {
+		settings.ClientURL = "{{ .ExternalURL }}"
+	}
+	if settings.Source == "" {
+		source, err := os.Hostname()
```
Comment (alexweav) [inline pkg/services/ngalert/notifier/channels/pagerduty.go:102]: Source is supposed to be the system that's affected by the firing alert - that's almost never going to be the machine running Grafana itself, but rather, whatever Grafana is monitoring. I'm not sure if hostname is the right default here.

Should the default simply be blank/not present?

**Review by alexweav — APPROVED** [review]
LGTM and nice to have overall. A couple minor comments about which default values to use, but overall looks good.

File: pkg/services/ngalert/notifier/channels/pagerduty.go (hunk 2 above)
Comment (yuri-tceretian) [reply pkg/services/ngalert/notifier/channels/pagerduty.go:102]: I made this for backward compatibility. I do not have a strong opinion, though.

Comment (yuri-tceretian) [reply pkg/services/ngalert/notifier/channels/pagerduty.go:102]: I think we can remove it later as part of the unificiation process

File: pkg/services/ngalert/notifier/channels/pagerduty.go (hunk 1 above)
Comment (yuri-tceretian) [reply pkg/services/ngalert/notifier/channels/pagerduty.go]: I think we will do this later as part of unification process.

**Pull request merged**